steps/measure.py

```diff
@@ -7,30 +7,34 @@
 def run_circuit_and_measure(
     backend_specs, circuit, noise_model="None", device_connectivity="None"
 ):
-    backend = create_object(json.loads(backend_specs))
-    circuit = load_circuit(circuit)
-
+    backend_specs = json.loads(backend_specs)
     if noise_model != "None":
-        backend_specs_dict["noise_model"] = load_noise_model(noise_model)
+        backend_specs["noise_model"] = load_noise_model(noise_model)
     if device_connectivity != "None":
-        backend_specs_dict["device_connectivity"] = load_circuit_connectivity(
+        backend_specs["device_connectivity"] = load_circuit_connectivity(
             device_connectivity
         )
+
+    backend = create_object(backend_specs)
+    circuit = load_circuit(circuit)
```

Comment:
`circuit = load_circuit(circuit)` looks confusing. I assume one of these is different from the other?
Also, we're missing typing for these functions.

Reply:
Added typing here: 8e003d3, the confusing nature of the variable names is because of workflow-v1 -> Can touch base with you about this if needed

Reply:
Can't we rename these variables to "circuit_str" and "circuit_obj" or sth more readable like this?

Reply:
The variable name is reflected in the workflow, so then we would have `circuit_str` in the workflow which is worse in my opinion.

Reply:
Ok, can we change the name in the python code, add a comment or make it very explicit in the docstrings?